x/evm/client/cli/query.go

```diff
@@ -479,3 +481,77 @@ func GetCmdEvent(queryRoute string) *cobra.Command {
 	flags.AddQueryFlagsToCmd(cmd)
 	return cmd
 }
+
+// GetCmdERC20Tokens returns the query to get the ERC20 tokens for a given chain
+func GetCmdERC20Tokens(queryRoute string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "erc20-tokens [chain]",
+		Short: "Returns the ERC20 tokens for the given chain",
+		Args:  cobra.ExactArgs(1),
+	}
+	tokenType := cmd.Flags().String("token-type", "", "the token type [external|internal]")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		clientCtx, err := client.GetClientQueryContext(cmd)
+		if err != nil {
+			return err
+		}
+
+		queryClient := types.NewQueryServiceClient(clientCtx)
+
+		var tokenTypeEnum types.TokenType
+		switch *tokenType {
+		case "":
+			tokenTypeEnum = types.Unspecified
+		case "internal":
+			tokenTypeEnum = types.Internal
+		case "external":
+			tokenTypeEnum = types.External
+		}
+
+		res, err := queryClient.ERC20Tokens(cmd.Context(),
+			&types.ERC20TokensRequest{
+				Chain: args[0],
+				Type:  tokenTypeEnum,
+			})
+		if err != nil {
+			return err
+		}
+
+		return clientCtx.PrintProto(res)
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdTokenDetails returns the query to get the details for an ERC20 token
+func GetCmdTokenDetails(queryRoute string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "token-details [asset]",
```

Comment:
lets have this query be able to retrieve the token details both by symbol and asset name. We could use the same scheme we use for the `token-address` query, where we either use the `--asset` or the `--symbol` flag.